src/aggregator/aggregator/leader_flush_mgr.go

```diff
@@ -181,20 +190,69 @@ func (mgr *leaderFlushManager) Prepare(buckets []*flushBucket) (flushTask, time.
 		}
 	}
 
-	durationSinceLastPersist := time.Duration(nowNanos - mgr.lastPersistAtNanos)
-	if mgr.flushedSincePersist && durationSinceLastPersist >= mgr.flushTimesPersistEvery {
-		mgr.lastPersistAtNanos = nowNanos
-		mgr.flushedSincePersist = false
-		flushTimes := mgr.prepareFlushTimesWithLock(buckets, allShards)
-		mgr.flushTimesManager.StoreAsync(flushTimes)
-	}
-
 	if !shouldFlush {
+		// NB: if enough time has passed since last flush time persist, run the
+		// persist by itself.
+		if shouldUpdateFlushTimes {
```

Comment:
seems like this would always happen when `flushTimesPersistEvery=0` and there is nothing to flush, leading to unnecessary saves.

Reply:
This will get stopped by the update flush task here: https://github.com/m3db/m3/blob/arnikola/update-flush/src/aggregator/aggregator/leader_flush_mgr.go#L220 but can make it more explicit